lua/entities/gmod_wire_expression2/core/egpfunctions.lua

```diff
@@ -934,6 +934,32 @@ e2function array wirelink:egpVertices( number index )
 	return {}
 end
 
+--------------------------------------------------------
+-- Object Type
+--------------------------------------------------------
+__e2setcost(1)
+
+e2function array wirelink:egpObjectTypes()
+	if not EGP:ValidEGP(this) then return {} end
+	if not this.RenderTable or #this.RenderTable == 0 then return {} end
+	local objs = {}
+	for _, v in pairs(this.RenderTable) do
+		objs[v.index] = EGP.Objects.Names_Inverted[v.ID] or ""
+	end
+	self.prf = self.prf + #this.RenderTable/3
```

Comment:
Are we certain this.RenderTable is sequential?

Reply:
I haven't checked that part, I just relied on how other EGP functions use length operator on this table too. So, I could assume it is sequential.
I had wanted to use `next` for even faster iteration.
But, I'm not confidentially 100% sure (as of this moment).

Reply:
Looks like it is